image-processing/UI.py

```diff
@@ -1,484 +1,202 @@
-                                    try:
-                                        baseline = float(baseline_input)    #got rid of divide by 2 for now
-                                        if baseline < 1:
-                                            print("Invalid input. Please ensure a positive number is entered.")
-                                            continue
-                                        break
-
-                                    except:
-                                        print("Invalid input. Please ensure a number is entered.")
-                                        continue
+import sys
+import warnings
+from linefinder import Linefinder
+warnings.filterwarnings("ignore")
 
+class UserInterface():
 
-                                if view_plot == 'yes':
-                                    print('Result for Sample:  {}'.format(scans_new[scan][0].stem))
+    def __init__(self):
+        # self.temp_path = r"C:\Users\Melissa\OneDrive - Lancaster University\University\Third Year\PHYS 355\Sample Data\Carbon Veil\Scans\75dpi" temporary path to avoid inputting all the time
 
-                                    if sampleType == 1:
-                                        finder.severity_carbon(baseline, grouping) 
+        print("*** Welcome to the User Interface for evaluating Machine Direction line severity in TFP nonwoven samples! *** \nPlease ensure sample scans are pre-arranged in folders, saved as .jpeg, and are of the same material type and areal weight class. You will be led through a series of prompts for which responses must be entered, before results are delivered. Type CTRL+C into the terminal to quit the program at any point. \n") #Type 'back' if you wish to return to a previous question and change your answer.
 
-                                    if sampleType == 2:
-                                        finder.severity_metal_coated(baseline, grouping)
+        self.path = Path(self.get_path()) 
+        self.material_types = {1: 'Carbon Veil', 2: 'Metal-Coated Carbon Veil'}  
+        self.files = []
 
-                                    name = str(scans_new[scan][0].stem)
-                                    finder.plot_nice(name = name)
+    def get_path(self):
+        response = input("Please specify the path to the folder containing the sample scan files. Enter 'help' for further explanation: ")
 
-                                elif view_plot == 'no':
-                                    print('Result for Sample: {}'.format(scans_new[scan][0].stem))
+        if response.lower() == 'help':
+            print("Using file explorer on Windows, open the folder containing the scans. In the navigation bar at the top, right click on the name of the folder, and then click 'copy address as text'. \nThe path should look something like 'C:\Documents\Bob\SampleScans\Carbon\LowAW'\n")
```

Comment:
thanks for changing it from my personal folders lmao

Reply:
HAHA sorry i thought i'd make the example a short path so plonked a Bob in there